infra/main.bicep

```diff
@@ -104,49 +104,41 @@ module machineLearningEndpoint './core/host/ml-online-endpoint.bicep' = {
   }
 }
 
-module userAcrRolePush 'core/security/role.bicep' =
-  if (!empty(principalId)) {
-    name: 'user-acr-role-push'
-    scope: rg
-    params: {
-      principalId: principalId
-      roleDefinitionId: '8311e382-0749-4cb8-b61a-304f252e45ec'
-      principalType: 'User'
-    }
+module userAcrRolePush 'core/security/role.bicep' = if (!empty(principalId)) {
```

Comment:
IMHO, I would just use the resource here w/o a module. I don't think we are getting that much from the `infra/core/security/role.bicep`.  And during deployment in ARM, each module will go as independent deployment instead of all together, so the module adds a few seconds.  So I don't see a strong reason to use a module here.

Reply:
Thanks for the feedback, but going to leverage the role module for now for consistency.